olap/src/aggregator/mod.rs

```diff
@@ -102,26 +110,103 @@ pub async fn run(
-            col("index"),
-            col("event_type"),
-            col("stream_cid"),
-            col("controller"),
-            col("event_cid"),
-            Expr::Cast(Cast::new(Box::new(col("data")), DataType::Utf8)).alias("data"),
-            col("previous"),
-        ])?;
+    run_continuous_stream(ctx, shutdown_signal).await?;
 
-    // TODO call this in a loop, see AES-291
-    process_feed_batch(ctx, conclusion_feed).await?;
+    Ok(())
+}
+
+async fn run_continuous_stream(
+    ctx: SessionContext,
+    shutdown_signal: impl Future<Output = ()>,
+) -> Result<()> {
+    let processor = ContinuousStreamProcessor::new(ctx).await?;
+    let mut shutdown_signal = Box::pin(shutdown_signal);
 
-    shutdown_signal.await;
+    loop {
+        tokio::select! {
+        _ = &mut shutdown_signal => {
+            debug!("Received shutdown signal, stopping continuous stream processing");
+            break;
+        }
+        result = processor.process_batch() => {
+            match result {
+                Ok(true) => {
+                    // Batch processed successfully, continue to next iteration
+                    continue;
+                }
+                Ok(false) => {
+                    // No more batches to process
+                    debug!("No more batches to process, continuous stream processing complete");
+                    break;
+                }
+                Err(e) => {
+                    error!("Error processing batch: {:?}", e);
+                    return Err(e);
+                }
+                }
+            }
+        }
+    }
     Ok(())
 }
 
+/// Represents a processor for continuous stream processing of conclusion feed data.
+struct ContinuousStreamProcessor {
+    ctx: SessionContext,
+    last_processed_index: Cell<u64>,
+}
+
+impl ContinuousStreamProcessor {
+    async fn new(ctx: SessionContext) -> Result<Self> {
+        Ok(Self {
+            ctx,
+            last_processed_index: 0.into(),
+        })
+    }
+
+    async fn process_batch(&self) -> Result<bool> {
+        // Fetch the conclusion feed DataFrame
+        // TODO : Is there a benefit to caching the DataFrame if it is fetched lazily? Shou;d we cache the query plan till the point of the filter instead of cachign the entire data frame?
+        let conclusion_feed = self
+            .ctx
+            .table(TableReference::full("ceramic", "v0", "conclusion_feed"))
+            .await?
+            .select(vec![
+                col("index"),
+                col("event_type"),
+                col("stream_cid"),
+                col("controller"),
+                col("event_cid"),
+                Expr::Cast(Cast::new(Box::new(col("data")), DataType::Utf8)).alias("data"),
+                col("previous"),
+            ])?;
+        let batch = conclusion_feed
+            .filter(col("index").gt(lit(self.last_processed_index.get())))?
+            .limit(0, Some(100))?;
+
+        let df = batch.cache().await?;
```

Comment:
A comment here that we need a cache because we do two passes over the data would be good. First pass is the normal process_feed_batch pass while the second pass is to get the highest index